lib/apiservers/engine/backends/container_proxy.go

```diff
@@ -169,28 +169,32 @@ func (c *ContainerProxy) Client() *client.PortLayer {
 //
 // returns:
 //	(containerID, containerHandle, error)
-func (c *ContainerProxy) CreateContainerHandle(imageID string, config types.ContainerCreateConfig) (string, string, error) {
-	defer trace.End(trace.Begin(imageID))
+func (c *ContainerProxy) CreateContainerHandle(vc *viccontainer.VicContainer, config types.ContainerCreateConfig) (string, string, error) {
+	defer trace.End(trace.Begin(vc.ImageID))
 
 	if c.client == nil {
 		return "", "", InternalServerError("ContainerProxy.CreateContainerHandle failed to create a portlayer client")
 	}
 
-	if imageID == "" {
+	if vc.ImageID == "" {
```

Comment:
I actually don't think a backward compatibility issue exists here, since this is the create path (by definition, all containers created with this new code are new containers).

But I do think one exists in the inspect path, so thanks for pointing that out. I'll see what I can come up with (maybe by bugging @emlin).